Microsoft.Toolkit.Mvvm/Messaging/StrongReferenceMessenger.cs

```diff
@@ -344,108 +297,137 @@ public void UnregisterAll<TToken>(object recipient, TToken token)
 
                 var key = new Recipient(recipient);
 
-                if (!mapping!.TryGetValue(key, out DictionarySlim<TToken, Action<TMessage>>? dictionary))
+                if (!mapping!.TryGetValue(key, out DictionarySlim<TToken, object>? dictionary))
                 {
                     return;
                 }
 
                 // Remove the target handler
-                if (dictionary!.Remove(token))
+                if (dictionary!.TryRemove(token, out _) &&
+                    dictionary.Count == 0)
                 {
-                    // Decrement the total count, as above
-                    mapping.TotalHandlersCount--;
-
                     // If the map is empty, it means that the current recipient has no remaining
                     // registered handlers for the current <TMessage, TToken> combination, regardless,
                     // of the specific token value (ie. the channel used to receive messages of that type).
                     // We can remove the map entirely from this container, and remove the link to the map itself
                     // to the current mapping between existing registered recipients (or entire recipients too).
-                    if (dictionary.Count == 0)
-                    {
-                        mapping.Remove(key);
+                    mapping.TryRemove(key, out _);
 
-                        HashSet<IMapping> set = this.recipientsMap[key];
+                    HashSet<IMapping> set = this.recipientsMap[key];
 
-                        set.Remove(mapping);
-
-                        if (set.Count == 0)
-                        {
-                            this.recipientsMap.Remove(key);
-                        }
+                    if (set.Remove(mapping) &&
+                        set.Count == 0)
+                    {
+                        this.recipientsMap.TryRemove(key, out _);
                     }
                 }
             }
         }
 
         /// <inheritdoc/>
-        public TMessage Send<TMessage, TToken>(TMessage message, TToken token)
+        public unsafe TMessage Send<TMessage, TToken>(TMessage message, TToken token)
```

Comment:
Does this `unsafe` impact anyone using this method?

Reply:
No, the `unsafe` block just enables us to use "unsafe" code (which doesn't really mean much at this point anyway, since you can do all sorts of ugly hacks using the `Unsafe` class and others, even though they don't require that), but it isn't part of the method signature. In this case it just makes the C# compiler happy when we do those `(void*)` casts down below 😄